exporter/awsemfexporter/grouped_metric.go

```diff
@@ -80,7 +80,7 @@ func addToGroupedMetric(pmd pmetric.Metric, groupedMetrics map[interface{}]*grou
 		}
 
 		// Extra params to use when grouping metrics
-		groupKey := groupedMetricKey(metadata.groupedMetricMetadata, labels)
+		groupKey := groupedMetricKey(metadata.groupedMetricMetadata, metadata.metricDataType, labels)
```

Comment:
I would consider removing the `metadata.metricDataType` and replace with `pmd.Type()` since the key does not contribute any value and increase memory resources created for the struct.